dotnet/src/Connectors/Connectors.Memory.Postgres/IPostgresVectorStoreCollectionSqlBuilder.cs

```diff
@@ -0,0 +1,134 @@
+    /// </summary>
+    /// <param name="schema">The schema of the table.</param>
+    /// <param name="tableName">The name of the table.</param>
+    /// <param name="properties">The properties of the table.</param>
+    /// <param name="keys">The keys of the records to get.</param>
+    /// <param name="includeVectors">Specifies whether to include vectors in the records.</param>
+    /// <returns>The built SQL command info.</returns>
+    PostgresSqlCommandInfo BuildGetBatchCommand<TKey>(string schema, string tableName, IReadOnlyList<VectorStoreRecordProperty> properties, List<TKey> keys, bool includeVectors = false) where TKey : notnull;
+
+    /// <summary>
+    /// Builds a SQL command to delete a record from the Postgres vector store.
+    /// </summary>
+    /// <param name="schema">The schema of the table.</param>
+    /// <param name="tableName">The name of the table.</param>
+    /// <param name="keyColumn">The key column of the table.</param>
+    /// <param name="key">The key of the record to delete.</param>
+    /// <returns>The built SQL command info.</returns>
+    PostgresSqlCommandInfo BuildDeleteCommand<TKey>(string schema, string tableName, string keyColumn, TKey key);
+
+    /// <summary>
+    /// Builds a SQL command to delete a batch of records from the Postgres vector store.
+    /// </summary>
+    /// <param name="schema">The schema of the table.</param>
+    /// <param name="tableName">The name of the table.</param>
+    /// <param name="keyColumn">The key column of the table.</param>
+    /// <param name="keys">The keys of the records to delete.</param>
+    /// <returns>The built SQL command info.</returns>
+    PostgresSqlCommandInfo BuildDeleteBatchCommand<TKey>(string schema, string tableName, string keyColumn, List<TKey> keys);
+
+    /// <summary>
+    /// Builds a SQL command to get the nearest match from the Postgres vector store.
+    /// </summary>
+    /// <param name="schema">The schema of the table.</param>
+    /// <param name="tableName">The name of the table.</param>
+    /// <param name="properties">The properties of the table.</param>
+    /// <param name="vectorProperty">The property which the vectors to compare are stored in.</param>
+    /// <param name="vectorValue">The vector to match.</param>
+    /// <param name="filter">The filter conditions for the query.</param>
+    /// <param name="skip">The number of records to skip.</param>
+    /// <param name="withEmbeddings">Specifies whether to include embeddings in the result.</param>
+    /// <param name="limit">The maximum number of records to return.</param>
+    /// <returns>The built SQL command info.</returns>
+    PostgresSqlCommandInfo BuildGetNearestMatchCommand(string schema, string tableName, IReadOnlyList<VectorStoreRecordProperty> properties, VectorStoreRecordVectorProperty vectorProperty, Vector vectorValue, VectorSearchFilter? filter, int? skip, bool withEmbeddings, int limit);
```

Comment:
Nit: other methods are using `includeVectors` instead of `withEmbeddings` for naming.